### PR TITLE
API Usage

### PR DESCRIPTION
**Pull request by shunjid**
This branch extends the application featuring the usage of REST API of **KonSchool**. For testing purpose, visit : https://analytics-konschool.azurewebsites.net/Analysis

### What's new
 - By clicking over any of the districts of the scalable vector graphics of Bangladesh's land area, users can view some analytical reports based on the clicked district
 - Usage of ChartJS for Male-Female ratio
 - Linear Regression on the key-parameter **area**
 - Finding 5 schools having less teachers through TSRatio
 - Google's material design with dark theme user-interface, minified scripts for production

### Long-Screenshot (Chrome-Dev-Tool)
analytics-konschool azurewebsites net_analysis(Laptop with MDPI screen) (1)

### CODE REVIEW

**Comment by shunjid** [comment]
**New upgrades**

- Removed **unused** CSS . **Minified** mapStyle.css and analytics.css
- Optimized Material Design Layout at **/Shared/_LayoutMaterial.cshtml**
- Fixed style-sheet issues detected by **[CodeFactor](https://www.codefactor.io/repository/github/maacpiash/konschool/pull/15)**
- Reduced razor page's (**/analysis**) size by loading svg of Bangladesh-Map from scripts

File: src/wwwroot/js/analytics.js [hunk 1]
```diff
@@ -21,11 +40,78 @@ function scrollToAnalyticalResultDivision(){
...
+    for(var i = data.length - 1; i >=0 ; i--){
+        filterMaleFemaleRatio(data[i]["MFRatio"], data[i]["type"]);
+    }
+
+    var mfPropertiesArray = $.map(maleFemaleRatio, function (v) { return v; });
+    createDoughnutBasedOnMFRatio(Object.keys(maleFemaleRatio), mfPropertiesArray, 'doughnut', $('#_maleFemaleCombinedDoughnut'), '');
+    summarizeMFinTable(data.length);
+}
+
+function filterMaleFemaleRatio(ratio, typeOfSchool) {
```
Comment (maacpiash) [inline src/wwwroot/js/analytics.js:149]: Now this function has brought an interesting revelation to me. I discovered that some schools were labeled co-ed or joint in the database although the MFRatio is 1.0 — some of these schools even have 'GIRLS HIGH SCHOOL' in their names!

I will fix these errors in the database as soon as I can. Then hopefully we won't have to see weird schools that are `typeBoysHasGirls` or `typeGirlsHasBoys` 🤣

Comment (shunjid) [reply src/wwwroot/js/analytics.js:149]: I was confused too during filtration of **MFRatio**. However, I thought, nothing is impossible and named such **cool** keys 😎😎

**Comment by maacpiash** [comment]
# What I liked ❤️ 

- The style: albeit incompatible with the rest of the web app, fantastic!
- The view components: the map SVG control and the charts are excellent!

# What I would like to see changed 🔧

The link in the header says 'API'. I think it should say 'Analysis'. Don't worry, I can take it from here.

# What I didn't understand 😕

Linear regression model. I didn't realize what it was predicting.

# Decision 💯

I liked what you did, @shunjid. I am going to merge this PR, and then work on fixing some minute issues.

**Comment by shunjid** [comment]
Thanks for merging this PR. In my next PR, I will be trying to:
- Add some notes on what I was trying to do with the linear regression
- Adding more statistical representation
- Last but not the least : usage of API with @tensorflow JS

**Pull request merged**